upxtreme: Add UP Xtreme support

mlkcampion opened the pull request:
UP Xtreme is based on the Intel(R) Core(TM) i3/i5/i7 Whiskey Lake SoCs.
The UP Xtreme presents one Raspberry Pi compatible HAT connector.

This implementaion supports i2c, spi, uart, adc and gpio through the
40pin HAT connector.

Gpio chardev capabilities have been disabled in this implementation. When
gpio chardev capabilities are enabled an input becomes unreadable after an
isr has been registered to the pin. See here for details:

https://github.com/intel-iot-devkit/mraa/issues/937

Tested on UP Xtreme, with UP Board Linux kernel 5.0.0
Features tested: gpio, gpio interrupts, i2c, spi, adc and uart.

Signed-off-by: Michael Campion <michael.campion@emutex.com>

mlkcampion committed:
upxtreme: Add UP Xtreme support

UP Xtreme is based on the Intel(R) Core(TM) i3/i5/i7 Whiskey Lake SoCs.
The UP Xtreme presents one Raspberry Pi compatible HAT connector.

This implementaion supports i2c, spi, uart, adc and gpio through the
40pin HAT connector.

Gpio chardev capabilities have been disabled in this implementation. When
gpio chardev capabilities are enabled an input becomes unreadable after an
isr has been registered to the pin. See here for details:

https://github.com/intel-iot-devkit/mraa/issues/937

Tested on UP Xtreme, with UP Board Linux kernel 5.0.0
Features tested: gpio, gpio interrupts, i2c, spi, adc and uart.

Signed-off-by: Michael Campion <michael.campion@emutex.com>

Propanu reviewed (approved):
This looks good, I don't have any objections whatsoever just two simple questions.

I see that you have added the option to force the `MRAA_UPXTREME` platform, does autodetection work correctly when this is not used?

Also, this requires a custom kernel like the other UP boards to use. Does the current Emutex PPA fully support this already? I do not see the board listed on the UP Community page yet. Unfortunately I can't test this either, as I don't have the board yet.

Thanks a lot for an otherwise great contribution to Eclipse MRAA!

mlkcampion committed:
upxtreme: Match UP Xtreme Platform enum value

After rebase UP Xtreme supported platform enum value was changed
to 24 in types.h, this commit updates the enum value in types.hpp
to this value.

Signed-off-by: Michael Campion <michael.campion@emutex.com>

mlkcampion committed:
upxtreme: add spdx tags to source files

Remove MIT notice and replace with SPDX tags in UP Xtreme sources.

Signed-off-by: Michael Campion <michael.campion@emutex.com>
